### DISCUSSION TITLE
Restore db from backup

### DISCUSSION THREAD

**Discussion by tvd0x2a** [craftcms/cloud#26] [Feature Requests & Enhancements]
We’d love to be able to restore a database from an existing backup in the console UI. Bonus feature would be restoring a non production environment from the production environment.

**Comment by jasonmccallister** [comment]
@tvd0x2a this discussion has been on our list for a while and definitely on the feature list. We are trying to work out any potential gotchas a head of time. Some of those questions are:

1. Are assets included or maybe an optional item when restoring one environment to another
2. Do we _only_ allow production to non-production or allow any environment to facilitate in place upgrades
3. Do environment variables also get copied

Lots of little details we want to ensure make it easy as possible and welcome any feedback and ideas!

**Reply by tvd0x2a** [reply]
I can see how there is more to consider here! I think for all 3 questions there's an element of giving Craft developers the ability to make the tradeoff that works for them and their needs, while picking a default that works for most.

Speaking for myself/the apps I work on:

> 1. Are assets included or maybe an optional item when restoring one environment to another

Optional would work for us. We would likely skip assets for testing simple or non-asset changes, since it might take a while to sync them. It would be nice to have the ability to opt into restoring assets to an environment when needed.

> 2. Do we only allow production to non-production or allow any environment to facilitate in place upgrades

I would like to be able to configure our org such that the production database can't be accidentally restored from a pre production environment. There might become a time when that is useful, and we would hopefully be able to temporarily disable that configuration. What if this was a project setting?

> 3. Do environment variables also get copied

I don't think we would want this for restoring our environments. I could see it being useful when creating an environment from scratch. What if it's optional, similar to syncing assets?

Thx!
